streaming/src/main/scala/org/apache/spark/streaming/util/StateMap.scala

```diff
@@ -61,8 +62,7 @@ private[streaming] object StateMap {
   def empty[K, S]: StateMap[K, S] = new EmptyStateMap[K, S]
 
   def create[K: ClassTag, S: ClassTag](conf: SparkConf): StateMap[K, S] = {
-    val deltaChainThreshold = conf.getInt("spark.streaming.sessionByKey.deltaChainThreshold",
-      DELTA_CHAIN_LENGTH_THRESHOLD)
+    val deltaChainThreshold = conf.get(SESSION_BY_KEY_DELTA_CHAIN_THRESHOLD)
```

Comment:
Is `DELTA_CHAIN_LENGTH_THRESHOLD` used other than this?

Reply:
Only here `DELTA_CHAIN_LENGTH_THRESHOLD`

Reply:
OK, then could we remove `DELTA_CHAIN_LENGTH_THRESHOLD`?

Reply:
OK.

Reply:
I'm sorry, I read it wrong.`DELTA_CHAIN_LENGTH_THRESHOLD` used in two place.
https://github.com/apache/spark/blob/5c29613428bffad5b0aae7cbcec85123a1ef4819/streaming/src/main/scala/org/apache/spark/streaming/util/StateMap.scala#L87
https://github.com/apache/spark/blob/5c29613428bffad5b0aae7cbcec85123a1ef4819/streaming/src/main/scala/org/apache/spark/streaming/util/StateMap.scala#L102

Reply:
Do the newly defined config's default value and the mentioned other places have to keep the same value? I mean if one change the one should the others be changed? (now the value defined in 2 distinct places)

Reply:
I'm not sure about this, I hope someone will be more clear about this.

Reply:
I've had a deeper look and `DELTA_CHAIN_LENGTH_THRESHOLD` is `OpenHashMapBasedStateMap` specific default value. I don't think we have to move it but I think the newly extracted configuration must use the constant.

Reply:
OK.